File: 7reverseInteger/reverseInteger.py
class Solution:
    def reverse(self, x: int) -> int:
        result = 0
        while x:
            if abs(result) > ((1 << 31) - 1) / 10: return 0
            else:
                result = result * 10 + int(math.fmod(x, 10))
                x = int(x / 10)
        return result

File: 7reverseInteger/reverseInteger.java
class Solution {
    public int reverse(int x) {
        var result = 0;
        while (x != 0)
            if (Math.abs(result) > Integer.MAX_VALUE / 10) return 0;
            else
            {
                result = result * 10 + x % 10;
                x /= 10;
            }
        return result;
    }
}
